### ISSUE TITLE
Extend vertical partitioned demonstration

### ISSUE THREAD

**Issue by TTitcombe** [OpenMined/PyVertical#25]
## Description
Extend the MVP (partitioning MNIST into images and labels) to work on arbitrary vertically partitioned datasets

## Why?
The dataset/dataloader/data partitioning/splitNN architecture/PSI functions are coded assuming the provided data is MNIST and the partitioning function split images and labels. Fortunately, the real world has more data than just MNIST. In this epic we will generalise the code to work with many datasets and partitions

## Breakdown
IN PROGRESS

## Who else?
May require work in PySyft and PSI

## Additional Context
Low priority.
Should be completed after #2 and #3
